File: scala/parsers/src/main/scala/jp/co/teruuu/parser/common/EOFParser.scala
package jp.co.teruuu.parser.common

class EOFParser() extends Parser[String] {

  override def parse(input: String): ParseResult[String] = {
    input.length == 0 match {
      case true => ParseOk("", "")
      case false => ParseNg("expect EOF:, actual " + input, input)
    }
  }
}

File: java/parsers/src/main/java/jp/co/teruuu/parser/common/EOFParser.java
package jp.co.teruuu.parser.common;

public class EOFParser implements Parser<String> {
    @Override
    public ParseResult<String> parse(String input) {
        if(input.length() != 0) {
            return new ParseResult.Failure<>("expected: EOF, actual: " + input.charAt(0), input);
        } else {
            return new ParseResult.Success<String>("", "");
        }
    }
}